src/data/repositories/DataElementD2GroupRepository.ts

```diff
@@ -0,0 +1,38 @@
+import { D2Api } from "../../types/d2-api";
+import { DataElementGroupRepository } from "../../domain/repositories/DataElementGroupRepository";
+import { DataElementGroup } from "../DataElementGroup";
+import { Code } from "../../domain/entities/Ref";
+import { D2DataElementGroup } from "./D2DataElementGroup";
+
+export class DataElementD2GroupRepository implements DataElementGroupRepository {
+    private d2DataElementGroup: D2DataElementGroup;
+
+    constructor(private api: D2Api) {
+        this.d2DataElementGroup = new D2DataElementGroup(this.api);
+    }
+
+    async getByCode(code: Code): Promise<DataElementGroup> {
+        const response = await this.d2DataElementGroup.getByIdentifiables([code]);
+        return response[0];
```

Comment:
I see that in this project we did not have the checked indexed access in tsconfig (`"noUncheckedIndexedAccess": true`), so TS does not validate it. But we should. Raise an exception if response[0] is empty.